### PR TITLE
Fixing logging of the correct address gRPC / headers

### PR DESCRIPTION
**Pull request by nikkolasg**

### CODE REVIEW

File: net/peer.go [hunk 1]
```diff
@@ -28,3 +36,57 @@ func CreatePeer(addr string, tls bool) Peer {
...
+		if str == "" {
+			return true
+		}
+		for _, r := range reserved {
+			if strings.HasPrefix(str, r) {
+				return true
+			}
+		}
+		return false
+	}()
```
Comment (Kubuxu) [inline net/peer.go:77]:
```
var lookAtHeader bool
if str == "" {
   lookAtHeader = true
} else {
   	for _, r := range reserved {
			if strings.HasPrefix(str, r) {
				lookAtHeader = true
        break
		}
}

Comment (Kubuxu) [reply net/peer.go:77]: IDK which one is cleaner.

File: net/peer.go [hunk 2]
```diff
@@ -28,3 +36,57 @@ func CreatePeer(addr string, tls bool) Peer {
...
+func RemoteAddress(c context.Context) string {
+	peer, ok := peer.FromContext(c)
+	var str string = ""
+	if ok {
+		str = peer.Addr.String()
+	}
+	// https://en.wikipedia.org/wiki/Reserved_IP_addresses
+	reserved := []string{
+		"192.168",
+		"10.0.0",
```
Comment (Kubuxu) [inline net/peer.go:61]: Whole `10.` is reserved for private network.

File: net/peer.go [hunk 3]
```diff
@@ -28,3 +36,57 @@ func CreatePeer(addr string, tls bool) Peer {
...
+		"127.0.0",
```
Comment (Kubuxu) [inline net/peer.go:62]: Same with `127.` is reserved for localhost.

File: net/peer.go [hunk 4]
```diff
@@ -28,3 +36,57 @@ func CreatePeer(addr string, tls bool) Peer {
...
+		"192.0.0",
+		"192.168",
```
Comment (Kubuxu) [inline net/peer.go:64]: `192.168` is repeated.

File: net/peer.go (hunk 1 above)
Comment (nikkolasg) [reply net/peer.go]: changed to your example - i dont like for loops in if/else but it's better than calling a function.. :D

File: net/peer.go [hunk 5]
```diff
@@ -28,3 +36,55 @@ func CreatePeer(addr string, tls bool) Peer {
...
+// ```
+//
+func RemoteAddress(c context.Context) string {
+	peer, ok := peer.FromContext(c)
+	var str string = ""
+	if ok {
+		str = peer.Addr.String()
+	}
+	// https://en.wikipedia.org/wiki/Reserved_IP_addresses
+	reserved := []string{
```
Comment (willscott) [inline net/peer.go:59]: you're missing 172.16.
more generally, can crib off of https://github.com/multiformats/go-multiaddr-net/blob/master/private.go#L11

**Pull request merged**